chain/network/src/time.rs

```diff
@@ -0,0 +1,99 @@
+#![allow(dead_code)]
+use once_cell::sync::Lazy;
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+// TODO: consider wrapping these types to prevent interactions
+// with other time libraries, especially the the direct access
+// to the realtime (i.e. not through the Clock).
+pub type Instant = time::Instant;
+pub type Utc = time::OffsetDateTime;
+pub type Duration = time::Duration;
```

Comment:
Question: do we want signed duration? Or do we want saturating duration? My gut feeling is that we actually want saturating one.

Reply:
I want it signed so that is behaves like a field within the range we operate on.